includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1235,6 +1235,10 @@ public function process_payment_for_order( $cart, $payment_information, $schedul
 				$request->set_payment_methods( $payment_methods );
 				$request->set_cvc_confirmation( $payment_information->get_cvc_confirmation() );
 
+				if ( Payment_Method::AFTERPAY === $this->get_selected_stripe_payment_type_id() && $this instanceof UPE_Split_Payment_Gateway ) {
+					$request->set_shipping( $this->get_shipping_data_from_order( $order ) );
+				}
```

Comment:
Just poking my nose in to mention that this field _was_ actually included in the original Affirm/Afterpay implementation here -> https://github.com/Automattic/woocommerce-payments/pull/6492/files#diff-bcfd303d507c33a27671935e724ee64d0df30ff2a18bb6ef4c94f28d48190124R1120. However, this specific piece of the implementation was inert (see #6792 for more details) and appeared to have no effect on completing transactions, so it was decided to remove it. 

I think it might not have been required with the legacy UPE element (used with both legacy UPE and split UPE), because when the payment element confirms the intent, Stripe tries its hardest to add any missing fields that might be required. That probably also explains this comment (here -> https://github.com/Automattic/woocommerce-payments/issues/6792#issuecomment-1640054759), which states that the billing address is used for this parameter, even when there is a separate shipping address, because the payment element doesn't know any better.

The deferred intent implementation is more manual, because we confirm the intent ourselves, rather than allowing the payment element to perform this task. I think it probably couldn't hurt to add this parameter for all UPE implementations, though it is only _required_ for the deferred intent UPE. 

I think checking whether the gateway is an instance of `UPE_Split_Payment_Gateway` is accurate enough, since the deferred intent UPE is presented to all split UPE users, as of the 6.4.0 release. Alternatively, you could just check `WC_Payments_Features::is_upe_deferred_intent_enabled()` to be as accurate and possibly a bit more explicit. But I'm not sure which conditional will honestly have greater longevity in the grand scheme of things, so no biggie. 🤷